lib/src/common/cache/cache.dart

```diff
@@ -294,6 +327,25 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
     return new Future.value();
   }
 
+  /// Run [callback] on [TValue] associated with [id].
```

Comment:
#nit this Is a bit vague for the first sentence of a Dart doc comment, and the second paragraph could be [condensed](https://www.dartlang.org/guides/language/effective-dart/documentation#prefer-brevity) a bit.
```dart
/// Returns `true` and calls [callback] if there is a cached, unreleased
/// [TValue] associated with [id]; returns `false` otherwise.
///
/// This does not perform a [get] or [getAsync], and as a result,
/// will not affect retention or removal of a
/// [TIdentifier][TValue] pair from the cache.
///
/// If the [Cache] [isOrWillBeDisposed] then a [StateError] is thrown.
```